bd-state/src/lib.rs

```diff
@@ -38,6 +38,107 @@ pub type TimestampedStateValue = (String, OffsetDateTime);
 /// A map of keys to timestamped values for a single scope.
 pub type ScopedStateMap = AHashMap<String, TimestampedStateValue>;
 
+//
+// StateChangeType
+//
+
+/// The type of state change that occurred.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StateChangeType {
+  /// A new key was inserted
+  Inserted { value: String },
+  /// An existing key was updated
+  Updated {
+    old_value: String,
+    new_value: String,
+  },
+  /// A key was removed
+  Removed { old_value: String },
+  /// No change occurred (e.g., setting same value)
+  NoChange,
+}
```

Comment:
Can you avoid string copies in here? Not sure how this is going to be used but since it's already evaluated with lifetimes maybe it's possible.